README.md

```diff
@@ -1,95 +1,108 @@
+*create cslib_ret 0
+```
 
-### Limitations
+- That's it! :partying_face:
 
-Sadly, cslib is not always going to be a one-size-fits all solution to your ChoiceScript woes. There are inherent limitations in ChoiceScript that cannot be overcome, no matter how much extra ChoiceScript we write. In addition, when writing a utility like cslib (that aims to appeal to a large number of people), certain tradeoffs have to be made. So, if you plan to use it, you should keep the following limitations in mind:
+You can now call any of the routines described in the file. :muscle:
 
-- **Performance**
-ChoiceScript isn't built for performance, and cslib isn't going to make that any better. The library relies heavily on usage of *gosub_scene, which is costly but it's also the only reliable way to handle complex variable scoping. If you're using the odd function or two as a utility here and there, then you shouldn't have any problems. If, however, you're planning on a super awesome complex simulator that loops through thousands of strings which it passes to cslib:replace or cslib:find... At that point, you might want to consider other options (including a different scripting language!).
+Routines that need to return a value will generally do so through the dedicated global variable: `cslib_ret`. So make sure to capture its value into another variable!
 
-- **\*temp scope**
-Due to the usage of \*gosub_scene, and inherent limitations in ChoiceScript, access to current game state stored in temporary (\*temp) variables from inside cslib functions isn't going to be possible. Most of the time this isn't a problem as you can (and should!) be passing your values as parameters. However, there are definitely some desirable use-cases that aren't currently possible. The best example is passing an "array" (mytemp_1, mytemp_2, mytemp_3) to a function. You can't reliably do this with parameters as you can't change the amount of parameters you're passing at runtime. You can get around it with globals (\*create'd) arrays by passing the variable "prefix" (e.g. 'myvar' in myvar_1, myvar_2) as an argument and de-referencing it via {} syntax (and providing a length). However, having to create every array in the global scope is far from ideal.
+Note that some routines will need you to create more global variables, or follow a naming convention, make sure you read the documentation on the routine before using it.
 
-## Technical Details
+Also, remember that ChoiceScript is by nature not performant, which is fine, given its primary purpose. This library aims to extend ChoiceScript's core functionality, but don't go too crazy with it. Okay? :sweat_smile:
 
-Below follow some extra technical details that will be beneficial to anyone wishing to learn more about the cslib internals, or plan to contribute to its source code.
+Credits are not mandatory but very much appreciated. :pray:
```

Comment:
Credit to whom? The "CSLIB Team" "CSLIB Contributors", maybe just ask to link back to the project?
I think crediting individuals could get a bit messy long-term.

Reply:
Yeah, I thought about the project itself because the project has a list of contributors.  I think it's good for the project visibility. I'll clarify that.